src/twisted/python/reflect.py

```diff
@@ -436,6 +436,13 @@ def safe_str(o):
             return o.decode('utf-8')
         except:
             pass
+    if not _PY3:
+        # On Python 2, attempt to encode a unicode representation
+        # first.
+        try:
+            return unicode(o).encode('ascii', 'backslashreplace')
+        except:
```

Comment:
I guess it's important to catch `KeyboardInterrupt`?  I did leave the bare `except:` for consistency's sake.